workers/deployment_manager/DeploymentManager/Log.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Improbable.Worker;
+
+namespace DeploymentManager
+{
+    public class Log
+    {
+        private static readonly Mutex logMutex = new Mutex();
+
+        private static StreamWriter logStream;
+
+        public static void Print(LogLevel level, string message, Connection connection = null)
+        {
+            logMutex.WaitOne();
+            var oldColor = Console.ForegroundColor;
+            var msg = $"{level}: {message}";
+            switch (level)
+            {
+                case LogLevel.Debug:
+                    Console.ForegroundColor = ConsoleColor.Cyan;
+                    Console.Out.WriteLine(msg);
+                    break;
+                case LogLevel.Info:
+                    Console.ForegroundColor = ConsoleColor.Gray;
+                    Console.Out.WriteLine(msg);
+                    break;
+                case LogLevel.Warn:
+                    Console.ForegroundColor = ConsoleColor.DarkYellow;
+                    Console.Out.WriteLine(msg);
+                    break;
+                case LogLevel.Error:
+                case LogLevel.Fatal:
+                    Console.ForegroundColor = ConsoleColor.DarkRed;
+                    Console.Error.WriteLine(msg);
+                    Environment.ExitCode = 1;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            Console.ForegroundColor = oldColor;
+
+            Debug.WriteLine(msg);
+
+            logStream?.WriteLine(Encoding.UTF8.GetChars(Encoding.UTF8.GetBytes(msg)));
+            logMutex.ReleaseMutex();
+
+            if (connection != null)
+            {
+                connection.SendLogMessage(level, DeploymentManager.WorkerType, message);
```

Comment:
Would it be worth limiting what we send based on the log level?

Reply:
maybe? right now we don't send too many messages anyway so don't think there is a point in doing that right now.